src/segment-loader.js

```diff
@@ -665,9 +673,11 @@ export default class SegmentLoader extends videojs.EventTarget {
       return;
     }
 
-    let isEndOfStream = detectEndOfStream(this.playlist_,
-                                          this.mediaSource_,
-                                          segmentInfo.mediaIndex);
+    let isEndOfStream = detectEndOfStream(
+      this.playlist_,
+      this.mediaSource_,
+      segmentInfo.mediaIndex
+    ) && !this.sourceUpdater_.updating();
```

Comment:
Should this condition be inside the `detectEndOfStream` function?

Reply:
I thought about it but decided against it, because it seemed weird to pass the source updater just so we could check updating in the function. Perhaps a new private function on the segment loader would make sense? `this.isEndOfStream_`?